### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Sites): the topology is generated by 1-hypercovers

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, we show that a Grothendieck topology on a category `C` (with `C : Type u` and `[Category.{v} C]`) is (obviously) generated by 1-hypercovers of size `max u v`.

In order to facilitate this, the type `Cover.Relation` is repackaged in a slightly different manner. Some proofs have been cleaned up. Several `dsimp` lemmas which had been manually added have been removed (and replaced by auto-generated ones).

---
- [x] depends on: #13004

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by joelriou** [comment]
> Is this ready for review?

Yes, now it is!

File: Mathlib/CategoryTheory/Sites/Plus.lean [hunk 1]
```diff
@@ -255,19 +255,10 @@ theorem plusMap_toPlus : J.plusMap (J.toPlus P) = J.toPlus (J.plusObj P) := by
...
-  · dsimp [diagram]
-    cases I
-    simp only [Category.assoc, limit.lift_π, Multifork.ofι_pt, Multifork.ofι_π_app,
-      Cover.Arrow.map_Y, Cover.Arrow.map_f]
-    rfl
-  · erw [Multiequalizer.lift_ι]
-    dsimp [Cover.index]
-    simp only [Functor.map_id, Category.comp_id]
-    rfl
+  refine' Multiequalizer.hom_ext _ _ _ (fun II => _)
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Sites/Plus.lean:258]:
```suggestion
  refine Multiequalizer.hom_ext _ _ _ (fun II => ?_)
```

File: Mathlib/CategoryTheory/Sites/OneHypercover.lean [hunk 2]
```diff
@@ -212,6 +212,52 @@ end
...
+    exact S.1.downward_closed g.hf _
+  · intro hf
+    exact Sieve.ofArrows_mk _ _ ({ hf := hf } : S.Arrow)
+
+lemma preOneHypercover_sieve₁ (f₁ f₂ : S.Arrow) {W : C} (p₁ : W ⟶ f₁.Y) (p₂ : W ⟶ f₂.Y)
+    (w : p₁ ≫ f₁.f = p₂ ≫ f₂.f) :
+    S.preOneHypercover.sieve₁ p₁ p₂ = ⊤ := by
+  ext Y f
+  dsimp
+  simp only [Sieve.top_apply, iff_true]
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Sites/OneHypercover.lean:247]:
```suggestion
  simp only [Sieve.top_apply, iff_true]
```

**Review by dagurtomas — APPROVED** [review]
The refactor of `Relation` seems very reasonable to me and one obtains the `IsGeneratedByOneHypercovers.{max u v}` easily.

File: Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean [hunk 3]
```diff
@@ -377,9 +358,7 @@ def ofι (I : MulticospanIndex.{w} C) (P : C) (ι : ∀ a, P ⟶ I.left a)
         | WalkingMulticospan.right b => ι (I.fstTo b) ≫ I.fst b
       naturality := by
         rintro (_ | _) (_ | _) (_ | _ | _) <;>
-          dsimp <;>
-          simp only [Category.id_comp, Category.comp_id, Functor.map_id,
-            MulticospanIndex.multicospan_obj_left, MulticospanIndex.multicospan_obj_right]
+          dsimp <;> simp
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean:361]: I noticed that this `dsimp` is unnecessary but removing it significantly slows down the proof.

**Comment by erdOne** [comment]
Am I correct that @dagurtomas thinks this is good to go?

**Comment by dagurtomas** [comment]
Yes, I think I already reviewed and approved?

**Comment by erdOne** [comment]
Okay just making sure. Thanks Joël and Dagur!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**